### PR TITLE
Enable scattering integers into decimal columns

### PR DESCRIPTION
**Pull request by brandon-b-miller**
Closes https://github.com/rapidsai/cudf/issues/8157

### CODE REVIEW

**Review by cwharris — APPROVED** [review]

File: python/cudf/cudf/core/column/decimal.py [hunk 1]
```diff
@@ -29,6 +29,11 @@ class DecimalColumn(ColumnBase):
     def __truediv__(self, other):
         return self.binary_operator("div", other)
 
+    def __setitem__(self, key, value):
+        if isinstance(value, np.integer):
```
Comment (shwina) [inline python/cudf/cudf/core/column/decimal.py:33]: Do we want to use `cudf.utils.dtypes.is_integer` here instead?

**Review by shwina — APPROVED** [review]
LGTM, minor suggestion that you can feel free to ignore.

File: python/cudf/cudf/core/column/decimal.py (hunk 1 above)
Comment (kkraus14) [reply python/cudf/cudf/core/column/decimal.py:33]: Is this special casing handling numpy scalars?

Comment (brandon-b-miller) [reply python/cudf/cudf/core/column/decimal.py:33]: yeah, this is just for numpy integers. Pyarrow doesn't like them for some reason.

```
>>> pa.scalar(1, type=pa.decimal128(1,0))
<pyarrow.Decimal128Scalar: Decimal('1')>
>>> pa.scalar(np.int64(1), type=pa.decimal128(1,0))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pyarrow/scalar.pxi", line 851, in pyarrow.lib.scalar
  File "pyarrow/error.pxi", line 107, in pyarrow.lib.check_status
pyarrow.lib.ArrowTypeError: int or Decimal object expected, got numpy.int64
```

**Review by kkraus14 — APPROVED** [review]

**Comment by kkraus14** [comment]
@gpucibot merge

**Pull request merged**